### ISSUE TITLE
Feature Request: External Login via OIDC

### ISSUE THREAD

**Issue by fir3wall** [projectsend/projectsend#610]
Hi,

We are aiming to use Project Send as file exchange method for our clients. Currently in last phase of testing and we discovered that the Social Login feature provided is handled currently via Google.

Would be great if that would make general for any OIDC (preferred) or OAUTH application. We are utilizing Keycloak who if integrated wia ex OIDC/OAUTH/SAML can provide external login screen, centralized user management or 2-factor authentication.

Current functionality is limited to one vendor where making it generic will allow other products to integrate into auth flow.

Aiming to support on Patreon as the product looks great, just maybe little bit documentation would be useful ex cannot find what versions of php and mysql (or mariadb) are supported! But great job!

Thanks
Dariusz

**Comment by kkplein** [comment]
Yes! This feature would be _very_ welcome for us as well.

**Comment by ignacionelson** [comment]
Thank you for your comments!!
Currently I'm doing a big refactor on PS.
To replace the current Google login system I've found hybridauth/hybridauth which seems to handle a LOT of providers. 2FA could be done with RobThree/TwoFactorAuth
I'm still still working on the basis of the app so adding these new libraries is not a priority, but the next version will include these functions without a doubt!

**Comment by magohl** [comment]
Agree that OIDC support would be a highly appreciated feature.

**Comment by styk-tv** [comment]
@ignacionelson any betas floating around that would have hybridauth working? I can help you confirm Keycloak OIDC setup.

**Comment by kkplein** [comment]
yes, we would like to offer testing-help as well. :-) (unfortunately perhaps: also running keycloak here...)

**Comment by sbernhard** [comment]
Ping :-) I'm also interested to connect projectsend with saml/OIDC. 
I'm not averse to pressing the "donate" button for that either. Are there any possibilities for a quick implementation?

**Comment by kilakewe** [comment]
Also interested!

**Comment by dromer** [comment]
Wait what, this is still not in?

It's a big show-stopper for me to not have support for either openicd or saml.
Rest of the projects looks pretty mature, but adding an external identity provider should be high prio I think.

**Comment by ignacionelson** [comment]
@dromer hybridauth can handle it but I had to leave out the set up since I didn't want to delay the release. It will be added in the future.

**Comment by dromer** [comment]
@ignacionelson thnx, will have a look what that entails exactly.

Good luck with the release, SSO is getting more and more important for groups trying to self-host :)

**Comment by MarcusGraetsch** [comment]
Is it foreseeable when this will be added?

**Comment by D4M13N-D3V** [comment]
please

**Comment by doctorfree** [comment]
Is `OIDC` support still on the table? If so, is there a planned release date or preview snapshot?

**Comment by reg0bs** [comment]
I came here to ask for the same thing as people before me :) 
Having OIDC (or SAML) to authenticate and possibly authorize and provision accounts would be great.

**Comment by gravelfreeman** [comment]
I would love the ability to use Authelia!

**Comment by manicmoose** [comment]
Add me to the list - would love to put this behind Authentik...

**Comment by LufinityLucas** [comment]
I'm on the waiting list as well! We will be using this great project to share files back and fourth between ourselves and our customers. Our customers can already sign in to our systems using OIDC (Keycloak). To have ProjectSend support OIDC is more or less a requirement for us. :D

**Comment by ignacionelson** [comment]
This issue is being closed as part of a bulk cleanup. ProjectSend has undergone significant changes since this was reported (including r1720 and r1994), and many older issues no longer apply to the current codebase.

If you are still experiencing this problem on the latest version, please open a new issue with updated details and we'll be happy to look into it. Thank you for your contribution!
